polkadot/node/network/statement-distribution/src/v2/mod.rs

```diff
@@ -693,15 +695,24 @@ pub(crate) async fn handle_active_leaves_update<Context>(
 			}
 		});
 
+		let maybe_claim_queue = fetch_claim_queue(ctx.sender(), new_relay_parent)
```

Comment:
Given that we pass per value (can't re-use the result), might make sense to move that call into `determine_groups_per_para`.

The result of request_availability_cores is used in `find_active_validator_state`, but wrongly. It uses `para_id` a function we should remove, as it makes no sense to either return the `ParaId` of the occupying para or the scheduled one.*)

Regardless, I would move the fetching code into the function that need the data. `handle_active_leaves_update` is already pages of code.

*) Note: This is obviously unrelated, so can be a separate PR, but it should be fixed for Coretime asap. @tdimitrov

Reply:
https://github.com/paritytech/polkadot-sdk/issues/3948